blog-pipeline/src/blog_runner/_runner.py

```
@@ -14175,6 +14175,68 @@ def _direct_hubspot_push(self, final_md: Path, project_id: str) -> dict:
+        # 1. Graphics: image-manifest.json exists with at least one cdn_url
+        manifest_path = self.output_dir / "deliverables" / "image-manifest.json"
+        if not manifest_path.exists():
+            failures.append("image-manifest.json missing — no graphics available for embedding")
+        else:
+            try:
+                manifest = json.loads(manifest_path.read_text())
+                images = manifest.get("images", [])
+                cdn_images = [img for img in images if img.get("cdn_url")]
+                if not cdn_images:
+                    failures.append("image-manifest.json has no images with cdn_url")
+            except Exception as e:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)